### ISSUE TITLE
Discovery of new devices does not work

### ISSUE THREAD

**Issue by guardiande** [mabunixda/node-red-contrib-alexa-home#53]
**Describe the bug**
I used to have the Alexa Local Node Red flow until it stopped working yesterday.

After some time I found this flow and understood that it seems to have to do with firmware changes on the Alexa side.

Here's my setup:

- Remote installation at my mother's place (which complicates debugging somewhat)
- 1 Sonos Beam (mostly used by my mother)
- 1 Alexa Echo Dot 2nd Gen.

History:
- I switched from Alexa Local to Alexa Home.
- Alexa Home Controller running on port 60000.
- Redirected port 80 to 60000 via iptables nat rule.
- Telnet to port 80 works as well as queries sent to the controller (setup.xml)
- I was able to discover devices yesterday and it worked.

Today:
- Today I wanted to complete the device discovery but it does not find any new device.
- Network trace looks promising: Sonos Beam talks to Raspberry on port 80.
- Lights API call lists new devices as well as old ones.
- Node Red debug log shows no obvious errors.
- Logging show that the controller handles setup requests.
- The old devices still work.

I tried to follow all advices from issue 16.

**To Reproduce**
- Create a new Alexa Home node.
- Give it a new unique name.
- Search for new devices in the Alexa App or in the Amazon's web GUI.
- Alexa does not find the new device.

**Expected behavior**
Alexa should find the newly created Alexa Home node.

**Screenshots**

**Desktop (please complete the following information):**
 - Alexa Home 1.1.3
 - Sonos Beam
- Amazon Echo Dot 2nd Gen.
 - Node 10.16.3
 - npm 6.9.0
 - No Docker

**Comment by guardiande** [comment]
Maybe a few more details:

I have the same setup at my own place and switched from Alexa Local to Alexa Home today. Up until now discovery works fine over here. Even though I have lots more of devices (166 vs. 20).

The number 20 caught my interest as I read somewhere that new hub spawn dynamically each 20 devices. Could that be a problem?

**Comment by godofglow** [comment]
same proble. Also with other Alexa nodes.

**Comment by guardiande** [comment]
A workaround seems to be to delete the alexa home node that don't get discovered and add them again. This sometimes does not work on first try.

**Comment by godofglow** [comment]
I tried the 'alexa home node' . Some (realy devices) was found, but the most are not be able to control. The rst of the devices was not found

**Comment by cymplecy** [comment]
Just tried this out on a Pi4 - using iptables to redirect port 80 - Alexa does not discover my device - it looks as if its trying but never suceeds
![image](https://user-images.githubusercontent.com/2357428/64508310-11340680-d2d5-11e9-8165-c92cf1c892f2.png)

**Comment by godofglow** [comment]
> 
> 
> Just tried this out on a Pi4 - using iptables to redirect port 80 - Alexa does not discover my device - it looks as if its trying but never suceeds
> ![image](https://user-images.githubusercontent.com/2357428/64508310-11340680-d2d5-11e9-8165-c92cf1c892f2.png)

Same here. I did the same test with the same result

**Comment by xilef111** [comment]
I have the same problem here. Alexa Controller configured to use Port 80 ans says OK, but no devices are dicovered if I start the search
![image](https://user-images.githubusercontent.com/55101855/64548616-48370600-d32f-11e9-8ddf-fe660b65daa4.png)

If you need some special logs, tell me where I can find them and I will post the here

**Comment by LakeCruiser** [comment]
Ok, I broke down and subscribed to the HA Cloud.  307 devices discovered!  $5 per month is a small price to pay for what I have been through the past two weeks! 🤯

**Comment by jpmrodrigues** [comment]
Hi, I have a similar problem. I have 25 devices on my node-red instance but when I request a discovery it only finds the same 3.

I also have a few of these messages in node-red debugging window: ""unknown alexa node of type lights was requested: 9d49c8da649938""

**Comment by guardiande** [comment]
Even with the workaround of deleting and re-adding nodes it still is unstable: Already discovered devices suddenly stop working and new devices cannot be discovered.

**Comment by guardiande** [comment]
This might be part of the problem: Alexa Home seems to advertise on different ports (60000 - 60002 in this case).

```
Sep 10 08:39:48 raspithree start[20250]: 10 Sep 08:39:48 - [info] [alexa-home-controller:Alexa Controller] 60000/GET -> /alexa-home/setup.xml
Sep 10 08:39:48 raspithree start[20250]: 10 Sep 08:39:48 - [info] [alexa-home-controller:Alexa Controller] 60001/GET -> /alexa-home/setup.xml
Sep 10 08:39:48 raspithree start[20250]: 10 Sep 08:39:48 - [info] [alexa-home-controller:Alexa Controller] 60000/GET -> /alexa-home/setup.xml
Sep 10 08:39:48 raspithree start[20250]: 10 Sep 08:39:48 - [info] [alexa-home-controller:Alexa Controller] 60002/GET -> /alexa-home/setup.xml
Sep 10 08:39:58 raspithree start[20250]: 10 Sep 08:39:58 - [info] [alexa-home-controller:Alexa Controller] 60000/GET -> /alexa-home/setup.xml
Sep 10 08:39:58 raspithree start[20250]: 10 Sep 08:39:58 - [info] [alexa-home-controller:Alexa Controller] 60001/GET -> /alexa-home/setup.xml
```

Not sure if this is related to spawning a new hub for every 20 devices.

**Comment by guardiande** [comment]
Ok, here's the proof:

```
Sep 10 18:12:54 raspithree start[11789]: 10 Sep 18:12:54 - [info] [alexa-home-controller:Alexa Controller] 0 - announcing: http://*:60000/alexa-home/setup.xml
Sep 10 18:12:54 raspithree start[11789]: 10 Sep 18:12:54 - [info] Assigning alexa-home nodes to this controller
Sep 10 18:12:54 raspithree start[11789]: 10 Sep 18:12:54 - [info] [alexa-home-controller:Alexa Controller] 1 - alexa-home - Starting SSDP
Sep 10 18:12:54 raspithree start[11789]: 10 Sep 18:12:54 - [info] [alexa-home-controller:Alexa Controller] 1 - announcing: http://*:60001/alexa-home/setup.xml
Sep 10 18:12:54 raspithree start[11789]: 10 Sep 18:12:54 - [info] [alexa-home-controller:Alexa Controller] 2 - alexa-home - Starting SSDP
Sep 10 18:12:54 raspithree start[11789]: 10 Sep 18:12:54 - [info] [alexa-home-controller:Alexa Controller] 2 - announcing: http://*:60002/alexa-home/setup.xml
```

I guess this does not go well together with Alexa expecting Hue bridge to be available on port 80. I cannot forwand port 80 to many ports.

**Comment by mabunixda** [comment]
When old devices do net get recognized: is there anything in the logs that there is already something listening on port xy? 

alexa-home starts per 25 nodes a seperate http-Hub because alexa devices ( at least till gen2 ) were not able to fetch larger responses. I gonna retest this with gen3 devices and also once again gen2 devices if that is the problem.

I still do not know how other fix those problems - i just found that some other nodes are fighting the same fight :-(

**Comment by jpmrodrigues** [comment]
I sure hope that this isn't the issue but I just read this in Philips Hue page:

**Comment by mabunixda** [comment]
Now i am also using gen2 echo dots ... 

But i just noticed that the alexa web interface declares my nodes as different items now. Donot know i just the interfaces and texts changed only :(

**Comment by Rmafive** [comment]
I'm using a gen 1, 2, and 3 Echo dot and seeing the same issue.

**Comment by guardiande** [comment]
Things are getting worse: Now most devices don't work and new devices do not get discovered.

Here's some network traces:

```
GET /alexa-home/setup.xml HTTP/1.1
Host: 192.168.47.36:60000
Connection: Keep-Alive
User-Agent: Android/7.1.2 UPnP/1.0 Cling/2.0

HTTP/1.1 200 OK
X-Powered-By: Express
Content-Type: application/xml; charset=UTF-8
Date: Tue, 10 Sep 2019 17:49:00 GMT
Connection: keep-alive
Transfer-Encoding: chunked

6f5
<?xml version="1.0"?>
<root xmlns="urn:schemas-upnp-org:device-1-0">
    <specVersion>
        <major>1</major>
        <minor>0</minor>
    </specVersion>
    <URLBase>http:&#x2F;&#x2F;192.168.47.36:60000</URLBase>
    <device>
        <deviceType>urn:schemas-upnp-org:device:Basic:1</deviceType>
        <friendlyName>Amazon-Echo-NodeRed-Bridge http:&#x2F;&#x2F;192.168.47.36:60000</friendlyName>
        <manufacturer>Royal Philips Electronics</manufacturer>
        <manufacturerURL>https://nitram.at</manufacturerURL>
        <modelDescription>Philips hue Personal Wireless Lighting</modelDescription>
        <modelName>Philips hue bridge 2012</modelName>
        <modelNumber>1100110011001100</modelNumber>
        <modelURL>http://meethue.com</modelURL>
        <serialNumber>93eadbeef14</serialNumber>
        <UDN>uuid:f6543a06-da50-11ba-8d8f-1939a6a8777ec9</UDN>
        <serviceList>
            <service>
                <serviceType>(null)</serviceType>
                <serviceId>(null)</serviceId>
                <controlURL>(null)</controlURL>
                <eventSubURL>(null)</eventSubURL>
                <SCPDURL>(null)</SCPDURL>
            </service>
        </serviceList>
        <presentationURL>index.html</presentationURL>
        <iconList>
            <icon>
                <mimetype>image/png</mimetype>
                <height>48</height>
                <width>48</width>
                <depth>24</depth>
                <url>hue_logo_0.png</url>
            </icon>
            <icon>
                <mimetype>image/png</mimetype>
                <height>120</height>
                <width>120</width>
                <depth>24</depth>
                <url>hue_logo_3.png</url>
            </icon>
        </iconList>
    </device>
</root>
```

Here Alexa asks for devices that don't seem to exist anymore:

```
GET /api/Hg9h6nijJEgbKhtCnehmE32yMHKYUcG8tlqag7bD/lights/a358c8d2c2ab78 HTTP/1.1
Content-Type: application/json
User-Agent: Dalvik/2.1.0 (Linux; U; Android 5.1.1; AEOBC Build/LVY48F)
Host: 192.168.47.36
Connection: Keep-Alive
Accept-Encoding: gzip

HTTP/1.1 502 Bad Gateway
X-Powered-By: Express
Date: Tue, 10 Sep 2019 17:49:05 GMT
Connection: keep-alive
Content-Length: 0
```

This is the query to enumerate lights, the answer (not shown) seems legit, all lights seem to be included:

```
GET /api/c6260f982b43a226b5542b967f612ce/lights HTTP/1.1
Content-Type: application/json
User-Agent: Dalvik/2.1.0 (Linux; U; Android 5.1.1; AEOBC Build/LVY48F)
Host: 192.168.47.36
Connection: Keep-Alive
Accept-Encoding: gzip
```

Here's an example of switching one of the few working lights on and off:

```
GET /api/Hg9h6nijJEgbKhtCnehmE32yMHKYUcG8tlqag7bD/lights/f980284087d708 HTTP/1.1
Content-Type: application/json
User-Agent: Dalvik/2.1.0 (Linux; U; Android 5.1.1; AEOBC Build/LVY48F)
Host: 192.168.47.36
Connection: Keep-Alive
Accept-Encoding: gzip

HTTP/1.1 200 OK
X-Powered-By: Express
Content-Type: application/json; charset=utf-8
Content-Length: 317
ETag: W/"13d-qDsWWlaa4FcuV0HEBoO8AsANX4k"
Date: Tue, 10 Sep 2019 17:50:12 GMT
Connection: keep-alive

{"state":{"on":false,"bri":254,"hue":0,"sat":254,"effect":"none","xy":[0,0],"ct":199,"alert":"none","colormode":"ct","mode":"homeautomation","reachable":true},"swupdate":{"state":"noupdates","lastinstall":"2019-09-10T17:50:12"},"type":"Extendedcolorlight","name":"T..r","modelid":"LTW011","swversion":"1.29.0_r21169"}GET /api/Hg9h6nijJEgbKhtCnehmE32yMHKYUcG8tlqag7bD/lights/f980284087d708 HTTP/1.1
Content-Type: application/json
User-Agent: Dalvik/2.1.0 (Linux; U; Android 5.1.1; AEOBC Build/LVY48F)
Host: 192.168.47.36
Connection: Keep-Alive
Accept-Encoding: gzip

HTTP/1.1 200 OK
X-Powered-By: Express
Content-Type: application/json; charset=utf-8
Content-Length: 317
ETag: W/"13d-qDsWWlaa4FcuV0HEBoO8AsANX4k"
Date: Tue, 10 Sep 2019 17:50:12 GMT
Connection: keep-alive

{"state":{"on":false,"bri":254,"hue":0,"sat":254,"effect":"none","xy":[0,0],"ct":199,"alert":"none","colormode":"ct","mode":"homeautomation","reachable":true},"swupdate":{"state":"noupdates","lastinstall":"2019-09-10T17:50:12"},"type":"Extendedcolorlight","name":"T..r","modelid":"LTW011","swversion":"1.29.0_r21169"}GET /api/Hg9h6nijJEgbKhtCnehmE32yMHKYUcG8tlqag7bD/lights/f980284087d708 HTTP/1.1
Content-Type: application/json
User-Agent: Dalvik/2.1.0 (Linux; U; Android 5.1.1; AEOBC Build/LVY48F)
Host: 192.168.47.36
Connection: Keep-Alive
Accept-Encoding: gzip

HTTP/1.1 200 OK
X-Powered-By: Express
Content-Type: application/json; charset=utf-8
Content-Length: 317
ETag: W/"13d-qDsWWlaa4FcuV0HEBoO8AsANX4k"
Date: Tue, 10 Sep 2019 17:50:12 GMT
Connection: keep-alive

{"state":{"on":false,"bri":254,"hue":0,"sat":254,"effect":"none","xy":[0,0],"ct":199,"alert":"none","colormode":"ct","mode":"homeautomation","reachable":true},"swupdate":{"state":"noupdates","lastinstall":"2019-09-10T17:50:12"},"type":"Extendedcolorlight","name":"T..r","modelid":"LTW011","swversion":"1.29.0_r21169"}GET /api/Hg9h6nijJEgbKhtCnehmE32yMHKYUcG8tlqag7bD/lights/f980284087d708 HTTP/1.1
Content-Type: application/json
User-Agent: Dalvik/2.1.0 (Linux; U; Android 5.1.1; AEOBC Build/LVY48F)
Host: 192.168.47.36
Connection: Keep-Alive
Accept-Encoding: gzip

HTTP/1.1 200 OK
X-Powered-By: Express
Content-Type: application/json; charset=utf-8
Content-Length: 317
ETag: W/"13d-qDsWWlaa4FcuV0HEBoO8AsANX4k"
Date: Tue, 10 Sep 2019 17:50:12 GMT
Connection: keep-alive

{"state":{"on":false,"bri":254,"hue":0,"sat":254,"effect":"none","xy":[0,0],"ct":199,"alert":"none","colormode":"ct","mode":"homeautomation","reachable":true},"swupdate":{"state":"noupdates","lastinstall":"2019-09-10T17:50:12"},"type":"Extendedcolorlight","name":"T..r","modelid":"LTW011","swversion":"1.29.0_r21169"}GET /api/Hg9h6nijJEgbKhtCnehmE32yMHKYUcG8tlqag7bD/lights/f980284087d708 HTTP/1.1
Content-Type: application/json
User-Agent: Dalvik/2.1.0 (Linux; U; Android 5.1.1; AEOBC Build/LVY48F)
Host: 192.168.47.36
Connection: Keep-Alive
Accept-Encoding: gzip

HTTP/1.1 200 OK
X-Powered-By: Express
Content-Type: application/json; charset=utf-8
Content-Length: 317
ETag: W/"13d-qDsWWlaa4FcuV0HEBoO8AsANX4k"
Date: Tue, 10 Sep 2019 17:50:12 GMT
Connection: keep-alive

{"state":{"on":false,"bri":254,"hue":0,"sat":254,"effect":"none","xy":[0,0],"ct":199,"alert":"none","colormode":"ct","mode":"homeautomation","reachable":true},"swupdate":{"state":"noupdates","lastinstall":"2019-09-10T17:50:12"},"type":"Extendedcolorlight","name":"T..r","modelid":"LTW011","swversion":"1.29.0_r21169"}PUT /api/Hg9h6nijJEgbKhtCnehmE32yMHKYUcG8tlqag7bD/lights/f980284087d708/state HTTP/1.1
Content-Type: application/json
User-Agent: Dalvik/2.1.0 (Linux; U; Android 5.1.1; AEOBC Build/LVY48F)
Host: 192.168.47.36
Connection: Keep-Alive
Accept-Encoding: gzip
Content-Length: 11

{"on":true}HTTP/1.1 200 OK
X-Powered-By: Express
Content-Type: application/json; charset=utf-8
Content-Length: 239
ETag: W/"ef-OMVAt0IuOVujCk+D2BMHkmYUQ50"
Date: Tue, 10 Sep 2019 17:50:13 GMT
Connection: keep-alive

[{"success":{"/lights/1/state/bri":254}},{"success":{"/lights/1/state/on":true}},{"success":{"/lights/1/state/hue":0}},{"success":{"/lights/1/state/sat":254}},{"success":{"/lights/1/state/xy":[0,0]}},{"success":{"/lights/1/state/ct":199}}]GET /api/Hg9h6nijJEgbKhtCnehmE32yMHKYUcG8tlqag7bD/lights/f980284087d708 HTTP/1.1
Content-Type: application/json
User-Agent: Dalvik/2.1.0 (Linux; U; Android 5.1.1; AEOBC Build/LVY48F)
Host: 192.168.47.36
Connection: Keep-Alive
Accept-Encoding: gzip

HTTP/1.1 200 OK
X-Powered-By: Express
Content-Type: application/json; charset=utf-8
Content-Length: 316
ETag: W/"13c-1sndmJzUsPeFYELpcd2Sb3DgwxA"
Date: Tue, 10 Sep 2019 17:50:13 GMT
Connection: keep-alive

{"state":{"on":true,"bri":254,"hue":0,"sat":254,"effect":"none","xy":[0,0],"ct":199,"alert":"none","colormode":"ct","mode":"homeautomation","reachable":true},"swupdate":{"state":"noupdates","lastinstall":"2019-09-10T17:50:13"},"type":"Extendedcolorlight","name":"T..r","modelid":"LTW011","swversion":"1.29.0_r21169"}PUT /api/Hg9h6nijJEgbKhtCnehmE32yMHKYUcG8tlqag7bD/lights/f980284087d708/state HTTP/1.1
Content-Type: application/json
User-Agent: Dalvik/2.1.0 (Linux; U; Android 5.1.1; AEOBC Build/LVY48F)
Host: 192.168.47.36
Connection: Keep-Alive
Accept-Encoding: gzip
Content-Length: 12

{"on":false}HTTP/1.1 200 OK
X-Powered-By: Express
Content-Type: application/json; charset=utf-8
Content-Length: 240
ETag: W/"f0-yXocErB71Nowa0fdqtF4ZtAHQQI"
Date: Tue, 10 Sep 2019 17:50:16 GMT
Connection: keep-alive

[{"success":{"/lights/1/state/bri":254}},{"success":{"/lights/1/state/on":false}},{"success":{"/lights/1/state/hue":0}},{"success":{"/lights/1/state/sat":254}},{"success":{"/lights/1/state/xy":[0,0]}},{"success":{"/lights/1/state/ct":199}}]
```

**Comment by guardiande** [comment]
Here are today's findings:

* I have 4 Alexa devices but only one seems to be able to initiate discoveries (verified by switching off all but one device and snooping on the network traffic).
* All tests were done after forgetting all devices through the web ui every time.
* When this one device is not switched on only my real Hue devices are found. This even works when all Alexa devices are switched off - no idea from where they are discovered. I guess it is some sort of caching.
* When I turn on this one special Alexa, forget all devices and rediscover, only a fraction of my devices get discovered but this happens even when Node Red isn't running. So there some caching going on here again.

This is really annoying... Thank you, Amazon. NOT.

**Comment by paulwhitmore** [comment]
I have updated nodejs and node-red to the latest versions, and updated alexa-home to version 1.1.3 and my Echo Dot Gen 2 will not discover any new devices. I guess I'm seeing the same problem as everyone else. Is this something that will be fixed - or have Amazon stuffed it up ?

**Comment by mabunixda** [comment]
I was quite busy the last days... 

IMO is that amazon deployed a change and this affects the current implementation and the node must be adapted to this change...

**Comment by guardiande** [comment]
@mabunixda Could assigning the Alexa devices randomly to the spawned hubs work as a workaround to overcome the device limit and hubs not running on port 80 problem?

Not the best thing but eventually all devices would be discoverable:

* Restart flows (devices get assigned to random hub and eventually to that running on port 80).
* Discover devices.
* Repeat until all devices got discovered.

This can of course be optimized by not assigning randomly.. The rotation could also be triggered by sending an event to the controller node.

**Comment by Jibbonator** [comment]
So Here is my Workaorunnd/ Fix for me

My System runs on IoBroker with a Node Red instance.
I Use this node Red Module : https://flows.nodered.org/node/node-red-contrib-amazon-echo
This Module works local without any Cloud Connection.

But i could not assign Port 80 to the "Amazon Echo Hub" from the new Module.
So i routed my Port 80 tcp and udp to another Port.(In my Case 1880, but you can choose whatever you want) In fact you should use another Port like 1890 because 1880 is the default Node Red Port.

Here´s an Manual for IP Table Routing:
https://o7planning.org/de/11363/redirect-port-80-443-auf-ubuntu-server-mit-iptables
If you want to save at the end again (Beacuse of TCP and UDP), this is the Command: sudo iptables-save

After this I created a "Amazon Echo Hub" ,set the Hub to the Routed Port and replaced my Nodes with the new ones.
Connected all Nodes to the Hub and deployed my Flow.

That´s it . After that I started the Discover and all Devices were found again.

Hope it helps some people.

**Comment by guardiande** [comment]
@Jibbonator How many Alexa nodes do you have in use?

**Comment by Jibbonator** [comment]
I have 12 nodes. If you want i could create more nodes and Test it tomorrow. Just say how much.

**Comment by guardiande** [comment]
That would be great! mabunixda stated that at least the older Echos have had problem with more than 25 devices. If it is not too much to ask it would be awesome to have something like 40 devices :-)

**Comment by Jibbonator** [comment]
@guardiande 
Hey my testing is done. I have two Echo Dot 2nd Gen.
I quadrupled my 11 Nodes and gave the Rest numbers(13-44). Created 44 Nodes in one Flow in total.
My Echo Dot 2nd Gen discovered 47 Devices (2 Dots and 1 TV extra) on first try.

I´ve tried a few commands an all worked flawlessly.
Like "Turn 13 on" and then "Turn 24 off". Same Action but differnet Numbers.

Hope it solved your question.

**Comment by egs2003** [comment]
@Jibbonator I don't understand your attempt. You are using a node from another developer. node-red-contrib-echo and node-red-contrib-home have nothing to do with each other? Or am I wrong?

**Comment by Jibbonator** [comment]
@egs2003 
That´s absolutely right. I just found a working solution to solve the\my Problem. I just wanted to share, that there is a working way. Most people don´t have enought time or WAF to wait until this Node is updated. :)

**Comment by guardiande** [comment]
@Jibbonator Thank you very much. It seems that the Amazon Echo node does not have the same limitation as this node has.

So it might be a workaround for people not having luck with this node for the moment.

**Comment by Gieskanne** [comment]
There is also a difference whether Node-RED is installed in stand-alone or used in ioBroker. In ioBroker I had no luck to run this node ("node-red-contrib-alexa-home") or even "node-red-contrib-alexa-local". But the "node-red-contrib-amazon-echo" node does work inside the Node-RED ioBroker adapter for me. That is why the solution by @Jibbonator should help most people. BTW: why are there multiple different nodes for the same purpose?

**Comment by egs2003** [comment]
It's great when people can be helped. However, we and the developer of this node want to find out why this node doesn't work. So it would be interesting to know what makes node...echo different from node...home.

**Comment by Jibbonator** [comment]
For everyone who needs help or want to talk about the solution to use
node-red-contrib-amazon-echo.
here is a Discord server, so this issue Topic is less full

And the Manual will be updated for better understanding on the Discord

https://discord.gg/Turn7Mm

**Comment by kas2** [comment]
Hi mabunixda,
I did enjoy your node for a few weeks until it broke recently
Please confirm if your are able/willing to fix it
Thanks again for this great work

**Comment by mabunixda** [comment]
Thanks for all the input. quite busy days on day-job ;-) 

I hope i can fix this sh*t again in the next days. Such kind of a problem already occured at least once in the past

**Comment by mabunixda** [comment]
right now i can reproduce 2 different ( or even 3 ) different behaviors:
* no discovery on node version < 10 
* discovery working on node version = 10 
* no output of nodes in node version = 10 ( #56 )

I am working on the problems

**Comment by mabunixda** [comment]
I hate that comments  automated close issue with fixxed keywords 👎 

Please retry your setups with version 1.2.0, a limitation on setting colors is still present! 
But on/off and dimming work as expected in my setup with node 0.20.8 and nodejs 10

**Comment by Tranquility76** [comment]
With 1.2.0 same issue - only 4 out of 15 devices are discovered. :-(

**Comment by mabunixda** [comment]
@Tranquility76 are you using iptables setup or changed the controller port?

With 1.2.0 you can setup the controller port with its controller configuration

**Comment by Tranquility76** [comment]
I never used iptables, always used the default controller port 80…

If I do the discovery I can see requests to the controller but it seems not all devices are requested when it comes to the details per device request.

**Comment by mabunixda** [comment]
@Tranquility76 are you using gen2 or gen3 device? can you verify which device is doing the discovery? an IP should be listed in the logs ( and yes, at the moment there are massive logs :( )

**Comment by Tranquility76** [comment]
I gave the „discover new devices“ command to a Echo Dot Gen3 and in the log I could see requests from the Echo Dot Gen3 AND two Echo Gen2 AND from my Echo Dot Gen2.

So all of my devices (Gen2 and Gen3) started the discovery.

Tried then the discvovery from my Alexa app – same result. App started discovery and all 4 Echo devices could be seen in the logs…



Von: Martin Buchleitner <notifications@github.com>
Gesendet: Donnerstag, 3. Oktober 2019 07:40
An: mabunixda/node-red-contrib-alexa-home <node-red-contrib-alexa-home@noreply.github.com>
Cc: Sebastian Probst <sebastian@probst-net.de>; Mention <mention@noreply.github.com>
Betreff: Re: [mabunixda/node-red-contrib-alexa-home] Discovery of new devices does not work (#53)


@Tranquility76<https://github.com/Tranquility76> are you using gen2 or gen3 device? can you verify which device is doing the discovery? an IP should be listed in the logs ( and yes, at the moment there are massive logs :( )

**Comment by Tranquility76** [comment]
Did another test – turned off the Gen3 Echo Dot and started discovery from the app:

All three Gen2 devices showed up in the log doing the discovery… (and did not find any new device – only the 4 already discovered)

Von: Martin Buchleitner <notifications@github.com>
Gesendet: Donnerstag, 3. Oktober 2019 07:40
An: mabunixda/node-red-contrib-alexa-home <node-red-contrib-alexa-home@noreply.github.com>
Cc: Sebastian Probst <sebastian@probst-net.de>; Mention <mention@noreply.github.com>
Betreff: Re: [mabunixda/node-red-contrib-alexa-home] Discovery of new devices does not work (#53)


@Tranquility76<https://github.com/Tranquility76> are you using gen2 or gen3 device? can you verify which device is doing the discovery? an IP should be listed in the logs ( and yes, at the moment there are massive logs :( )

**Comment by kas2** [comment]
Hi
I still cannot discover devices using default 60000 port

>>All three Gen2 devices showed up in the log doing the discovery
Which log file are you referring to and how to access it ??
Thanks

**Comment by all-solutions** [comment]
Hi all,
I think at the moment there is no solution for that. At the moment Amazon is playing around every day with an other "feature". I have Hue and WEMO devices in my configuration. Today all devices with umlauts have only hieroglyphs in the name. I've send a message to Amazon and received that answer:

> Guten Tag Herr XXXXXX,
> 
> vielen Dank für Ihr Schreiben.
> 
> Ich habe Ihre Anfrage geprüft und kann Ihnen folgende Informationen mitteilen:
> 
> In den letzten Tagen hatten wir verschiedene Schwierigkeiten mit Nutzung von Alexa Services, weil derzeit unsere Technikern auf Entwicklung einer neue Software arbeiten und verschiedene Tests durchführen.  Dieser Software soll Ihr Erlebnis bei Nutzung von Echo und verschiedene Funktionen verbessern.
> 
> Ich habe eine Kopie Ihrer E-Mail an die zuständigen Mitarbeiter weitergeleitet, die sämtliche Vorschläge, Anregungen und Kommentare von unseren Kunden lesen und in regelmäßigen Abständen auswerten. Nur so wissen wir, was unsere Kunden wünschen und können unseren Service und unser Angebot ständig verbessern.

[english translation]

> Hello Mr. XXXXXX,
> 
> Thank you very much for your letter.
> 
> I have checked your request and can provide you with the following information:
> 
> In the last few days we have had several difficulties with using Alexa Services because our technicians are currently working on the development of new software and performing various tests.  This software is designed to improve your experience using Echo and various features.
> 
> I have forwarded a copy of your email to the appropriate staff who will read and periodically review any suggestions, suggestions and comments from our customers. This is the only way we know what our customers want and can constantly improve our service and offer.

**Comment by jaburges** [comment]
Could someone please simplify the issue?
I'm using a node-red docker, so can have docker forward the ports, but cant figure out from the thread if Alexa is looking at 80, or is node-red needing 80?

**Comment by mabunixda** [comment]
Nodered runs Default in Port 1880, Alexa works with the current Firmware best, when the Port 80 ist used for Alexa. You can configure this in the nodered node.
But when using docker you must map the correct Ports of course

**Comment by jaburges** [comment]
Thanks - although 80 doesn’t work either.

1880 is the Node-red UI
80 internal port : 80 external port?
(Or something different? No luck with any)
And in host network_mode, same issue?

**Comment by mabunixda** [comment]
I am using this to run a local Dev Version
https://github.com/mabunixda/node-red-contrib-alexa-home/blob/709a6c0012e94287062c380ddc20aa614a850258/scripts/run.sh#L3

Because of the Udp ssdp used in Alexa you must use Host Mode for Network.
Otherwise this also does Not require a Port mapping

⁣BlueMail for Android herunterladen ​

**Comment by Tranquility76** [comment]
Even with Port 80 configured right it might still not work.

There is still this open issue that since a few month Alexa is not able to detect (all) devices with this node.

In my case only 4 out of 15 devices are discovered – all the fixes provided in the last 2 months did not solve this issue… ☹

So maybe you configuration is already right but your are hitting this discovery issue too…

Von: jaburges <notifications@github.com>
Gesendet: Sonntag, 10. November 2019 09:32
An: mabunixda/node-red-contrib-alexa-home <node-red-contrib-alexa-home@noreply.github.com>
Cc: Sebastian Probst <sebastian@probst-net.de>; Mention <mention@noreply.github.com>
Betreff: Re: [mabunixda/node-red-contrib-alexa-home] Discovery of new devices does not work (#53)


Thanks - although 80 doesn’t work either.

1880 is the Node-red UI
80 internal port : 80 external port?
(Or something different? No luck with any)
And in host network_mode, same issue?

**Comment by jaburges** [comment]
any definitive way to test?
i have tried net=host
AND
exposed 80 & 60000 (neither of which i had to do before)

is this likely something that can be fixed?

**Comment by mabunixda** [comment]
@jaburges can you post your docker run commandline and flow? alexa and node-red are in the same lan ?

@Tranquility76 yeah it is hard to follow this issues if i am not able to reproduce the issue and there is no other dev available which is able to shed more light into that issue :-( 

I know that this is frustrating situation - also on my end as developer of this module

**Comment by Tranquility76** [comment]
Hi Martin,

I really appreciate your work and I guess it is not easy to hold the speed Amazon is changing things.

I am curious – what is https://github.com/datech/node-red-contrib-amazon-echo doing different? Because this node works…

**Comment by jaburges** [comment]
@mabunixda 
docker:

> sudo docker run -it -p 1880:1880 -p 60000:60000 -p 80:80 --user root --name nodered --restart always -v /home/name/docker/node-red/data:/data nodered/node-red-docker:0.20.8-v8
or
`sudo docker run -it --net host --user root --name nodered --restart always -v /home/name/docker/node-red/data:/data nodered/node-red-docker:0.20.8-v8`

flow is just alexa controller tried with `80` or `60000`
and node into a debug node (full msg) - but devices are discovered yet by Alexa

**Comment by jaburges** [comment]
> Hi Martin, I really appreciate your work and I guess it is not easy to hold the speed Amazon is changing things. I am curious – what is https://github.com/datech/node-red-contrib-amazon-echo doing different? Because this node works…

i've just tried this node and it has the same issue.
I've also tried:

> sudo docker run -it -p 1880:1880 -p 80:80 --user root --name nodered --restart always -v /home/name/docker/node-red/data:/data nodered/node-red:latest

Same problem :(

**Comment by Tranquility76** [comment]
I am running https://github.com/datech/node-red-contrib-amazon-echo on a Raspi and it just works.

All(!) devices are discovered and react on commands as expected. (Although the node output is slightly different so I had to modify my flows a little bit.)

**Comment by jaburges** [comment]
are your devices new (port 80) or older, and are you running node-red in a docker?

**Comment by Tranquility76** [comment]
Gen2 and Gen3 devices and as written running on a Raspi and not in docker.



Am 17.11.2019 um 10:46 schrieb jaburges <notifications@github.com>:

﻿

are your devices new (port 80) or older, and are you running node-red in a docker?

**Comment by jaburges** [comment]
switched to https://github.com/datech/node-red-contrib-amazon-echo
and that DOES work now on `network_mode=host` 
with `nodered/node-red:latest`

**Comment by mabunixda** [comment]
closing this one - either you moved to another node or nobody posted problem updates since long time
